### PR TITLE
Followup Enchancements to #224

### LINKED ISSUE #224: Implement Data Analysis page
**Describe the feature that you would like added**
Currently, the Data Analysis button in the main experiment UI outlined in ```/experiment_pages/experiment/data_analysis_ui.py``` renders a blank page. This page should instead allow the user to export an experiment's data as a common file type for data processing and analysis (.csv, .xls, etc.)

**Tasks that need to completed for this feature**
- [ ] Rename the Data Analysis button and Page headers to reflect the true purpose of the page, data exporting.
- [ ] Implement exporting in the Data Analysis Page
OR
- [ ] Make a new button/function for data exporting with included documentation
- [ ] Hide Data Analysis from main experiment UI while unused

**Additional context**
Mouser was originally conceived to include some more advanced data analysis options and tools, which is why the page exists in the first place. The current direction of the project does not include sophisticated data analysis, but should support common file types to be compatible with popular data analysis software.

### PR DESCRIPTION
**Pull request by moshernoah**
Fixes #224 

**What was changed?**

Comments were left on the other branch implementing fixes for issue #224 that are addressed here. Now, the user can select a folder location the same as if they were saving a file during experiment creation before exporting all saved data to .csv format.

**Why was it changed?**

This was changed as there were concerns moving forward about having the project save files to an internal folder that may not exist when the .exe changes are complete. This allows for portability with the data after it is no longer being launched from the command line. 

**How was it changed?**

The user is prompted within the data_analysis_ui.py file to select where they would like the exported data to be saved. Inside the experiment_database.py file the export_data_to_csv method has been edited to require a selected folder as an input argument. Should the user exit out of the folder selection method, the function is returned and the program continues to function as if the button was never pressed. If a folder was selected, it will create a subfolder holding the name of the experiment, with all exported files within it.

### CODE REVIEW

File: experiment_pages/experiment/data_analysis_ui.py [hunk 1]
```diff
@@ -27,10 +27,15 @@ def export_to_csv(self):
         if not self.db_file or not os.path.exists(self.db_file):
             print(f"Database file {self.db_file} not found.")
             return
+        
+        # Ask user for the directory to save CSV files
```
Comment (ikardasevic) [inline experiment_pages/experiment/data_analysis_ui.py:31]: Looks good and efficient!

File: databases/experiment_database.py [hunk 2]
```diff
@@ -447,22 +447,16 @@ def close(self):
...
-        # Create directory to save CSV files if it doesn't exist
-        if not os.path.exists(export_dir):
-            os.makedirs(export_dir)
-
         # Get the base name of the database file (without extension)
         db_name = os.path.splitext(os.path.basename(self.db_file))[0]
 
         # Create a directory with the database name to save CSV files
-        export_dir = os.path.join("exports", db_name)
+        export_dir = os.path.join(export_dir, db_name)
```
Comment (ikardasevic) [inline databases/experiment_database.py:456]: Good code and comments, makes sense.

File: databases/experiment_database.py [hunk 3]
```diff
@@ -447,22 +447,16 @@ def close(self):
         '''Closes database file.'''
         self._conn.close()
 
-    def export_all_tables_to_csv(self, export_dir="exports"):
+    def export_all_tables_to_csv(self, export_dir):
         '''Exports all tables in the database to separate CSV files.'''
-        # Create directory to save CSV files if it doesn't exist
-        if not os.path.exists(export_dir):
-            os.makedirs(export_dir)
-
```
Comment (ikardasevic) [inline databases/experiment_database.py:455]: Good job! This resolves the issue you told me about.

**Review by jackbelyeu — APPROVED** [review]
Just tested the export on the .exe and it worked. Nice changes.

**Pull request merged**